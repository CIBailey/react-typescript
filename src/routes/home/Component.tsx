import * as React from 'react';

// typing
import { RouteComponentProps } from 'react-router';

// module
import { Row, Col, Button } from 'reactstrap';

// sass import
import * as styles from './Styles.scss';

// interface props
export interface Props extends RouteComponentProps {
  push(to: string): void;
}

// interface state
interface State {
  hello: string;
}

export default class Home extends React.Component<Props, State> {
  // default props
  static defaultProps = {
    push: () => {}
  };

  constructor(props: Props) {
    super(props);
    this.state = {
      hello: 'Hello !'
    };
  }

  render() {
    return (
      <Row className={styles.container}>
        <Col lg="12" className={styles.container_button}>
          {this.state.hello}
        </Col>
        <Col lg="12" className={styles.container_button}>
          <Button onClick={() => this.setState({ hello: 'Hello world !' })} color="primary">
            Trigger state
          </Button>
        </Col>
        <Col lg="12" className={styles.container_button}>
          <Button onClick={() => this.props.push('/counter')} color="info">
            Go to counter
          </Button>
          <Button onClick={() => this.props.push('/translate')} color="info">
            Go to translate
          </Button>
        </Col>
      </Row>
    );
  }
}
