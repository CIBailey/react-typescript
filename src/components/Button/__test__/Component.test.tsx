import * as React from 'react';
import * as ReactTestRenderer from 'react-test-renderer';

import * as enzyme from 'enzyme';
import * as Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

// tslint:disable-next-line:import-name
import Component from '../index';

const Test = (
  <Component color={'info'} onClick={() => console.log('ty')}>
    <span>test</span>
  </Component>
);

describe('BUTTON COMPONENT', () => {
  const component: ReactTestRenderer.ReactTestRenderer = ReactTestRenderer.create(Test);

  it('checking snapshot', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Check if children is created', () => {
    expect(enzyme.shallow(Test).contains(<span>test</span>)).toBeTruthy();
  });

  it('Test props', () => {
    expect(enzyme.shallow(Test).prop('color')).toEqual('info');
  });

  it('simulate button', () => {
    enzyme
      .mount(Test)
      .find('button')
      .simulate('click');
  });
});
