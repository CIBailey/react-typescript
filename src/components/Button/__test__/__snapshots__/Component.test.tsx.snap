// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BUTTON COMPONENT checking snapshot 1`] = `
<button
  className="btn btn-info"
  onClick={[Function]}
  style={Object {}}
  type="button"
>
  <span>
    test
  </span>
</button>
`;
